### PR TITLE
Skip leading whitespace when detecting JSON writes without content-type

### LINKED ISSUE #2834: [0.9.0-rc32] Unable to write any data via console / API: "database is required"
Brand new install of rc32. I have created a database `test`. Attempting to do the following results in a 400 with the error "database is required":

```
$ curl -XPOST 'http://myserver:8086/write' -d '
{
    "database": "test",
    "retentionPolicy": "default",
    "time": "2015-05-22T16:42:01.234Z",
    "precision": "ms",
    "points": [
        {
            "name": "testing",
            "tags": {
                "host": "server01",
                "region": "us-west"
            },
            "fields": {
                "value": 0.64
            }
        }
    ]
}'
```

Strangely, and maybe related, in the CLI, I see this:

```
> show databases
name: databases
---------------
name
test

> use test
Using database test
> show series
ERR: database not found: test
> show retention policies
ERR: error parsing query: found EOF, expected identifier at line 1, char 25
> show measurements
ERR: database not found: test
````

### PR DESCRIPTION
**Pull request by jwilder**
Fixes #2834

### CODE REVIEW

File: services/httpd/handler.go [hunk 1]
```diff
@@ -397,9 +397,21 @@ func (h *Handler) writeError(w http.ResponseWriter, result influxql.Result, stat
...
+		var i int
+		for {
+			// JSON requests must start w/ an opening bracket
+			if body[i] == '{' {
+				h.serveWriteJSON(w, r, body, user)
+				return
+			}
+
+			// check that the byte is in the standard ascii code range
+			if body[i] > 32 && body[i] < 127 {
```
Comment (otoolep) [inline services/httpd/handler.go]: Why are you allowing values >= 127 to go through? Are they valid trailing characters for JSON?

Comment (jwilder) [inline services/httpd/handler.go]: Updated.  It breaks out of the loop and let the line parser handle it from here.

File: services/httpd/handler.go [hunk 2]
```diff
@@ -397,9 +397,21 @@ func (h *Handler) writeError(w http.ResponseWriter, result influxql.Result, stat
...
+	if len(body) > 0 {
+		var i int
+		for {
+			// JSON requests must start w/ an opening bracket
+			if body[i] == '{' {
+				h.serveWriteJSON(w, r, body, user)
+				return
+			}
+
+			// check that the byte is in the standard ascii code range
```
Comment (otoolep) [inline services/httpd/handler.go]: Not sure if this comment is accurate. More like skip any whitespace, or whitespace like characters?

**Comment by otoolep** [comment]
Makes general sense to me, though I do think the check might be a bit broad. It allows lots of characters other than whitespace. +1 if you are sure this is the right thing to do.

**Comment by toddboom** [comment]
+1

**Pull request merged**